dbt/include/bigquery/macros/materializations/table.sql

```diff
@@ -109,6 +109,15 @@ else:
 df.write \
   .mode("overwrite") \
   .format("bigquery") \
-  .option("writeMethod", "direct").option("writeDisposition", 'WRITE_TRUNCATE') \
+  .option("writeMethod", "indirect").option("writeDisposition", 'WRITE_TRUNCATE') \
+  {%- if partition_config.data_type | lower in ('date','timestamp','datetime') %}
```

Comment:
Have you considered breaking out the partition and cluster options as their own variables prior to inserting them into the `df.write` flow? I think it would make it more readable, especially given that we can't indent on python models.